native-gradle-plugin/src/main/java/org/graalvm/buildtools/gradle/NativeImagePlugin.java

```diff
@@ -134,6 +134,8 @@ private void configureJavaProject(Project project, Provider<NativeImageService>
                 project.getExtensions().findByType(JavaApplication.class).getMainClass()
         ));
 
+        project.getPlugins().withId("java-library", p -> buildExtension.getSharedLibrary().convention(true));
```

Comment:
You can set the convention to `false` in the `NativeImageOptions` constructor, which avoids the `@Optional` and `getOrElse` when you consume the value.